### ISSUE TITLE
[BUG] Mixed Nulls cannot deal with category dtypes

### ISSUE THREAD

**Issue by chendingyan** [deepchecks/deepchecks#1674]
**Describe the bug**
Mixed Nulls check cannot work on "category" dtypes

**To Reproduce**
1. Create a dataframe with some columns with "category" dtypes
<img width="977" alt="image" src="https://user-images.githubusercontent.com/16874978/174963320-a01274f6-5c08-463f-bf8c-cdf881aa88fd.png">
2. Then use mixed nulls check with default condition
<img width="860" alt="image" src="https://user-images.githubusercontent.com/16874978/174963461-95d0561e-880e-43b1-9eb9-57b2b89dc369.png">

**Expected behavior**
<img width="731" alt="image" src="https://user-images.githubusercontent.com/16874978/174963717-fa314603-bb75-44ba-8bc7-c47d6dce3ddc.png">

**Environment (please complete the following information):**
 - OS: Mac
 - Python Version:3.8
 - Deepchecks Version:0.8.0

**Additional context**

**Comment by matanper** [comment]
Hey, thanks for reporting the bug!

The problem is when using pandas `value_counts` on a column with category dtype, it returns all the categories even if their count is 0.

**Comment by chendingyan** [comment]
Yep, and also sometimes we are required to input data with category dtype while using deepchecks _model_evaulation_ checks. For example, we have a pre-trained LGB model with category features, and if we don't input data with category dtype columns, model.predict() will raise an error.
